diff --git "a/D:\\source\\GitHub\\lua\\lstrlib.c" "b/D:\\source\\GitHub\\cpp_static_embeded_lua\\lua_dirty\\lstrlib.c"
index 84b6e4e..5c3495c 100644
--- "a/D:\\source\\GitHub\\lua\\lstrlib.c"
+++ "b/D:\\source\\GitHub\\cpp_static_embeded_lua\\lua_dirty\\lstrlib.c"
@@ -25,7 +25,8 @@
 #include "lauxlib.h"
 #include "lualib.h"
 
-
+#include "encode_string.h"
+ 
 /*
 ** maximum number of captures that a pattern can do during
 ** pattern-matching. This limit is arbitrary, but must fit in
@@ -1715,6 +1716,7 @@ static const luaL_Reg strlib[] = {
   {"pack", str_pack},
   {"packsize", str_packsize},
   {"unpack", str_unpack},
+  {"mbcs2utf8",str_mbcs2utf8},
   {NULL, NULL}
 };
 
